src/binaryutils.cpp

```diff
@@ -84,6 +84,12 @@ cv::Mat convertStringValToMat(const std::string& stringVal) {
     return cv::imdecode(dataMat, cv::IMREAD_UNCHANGED);
 }
 
+std::string convertMatToStringVal(const cv::Mat& dataMat) {
+    std::vector<unsigned char> buf;
+    cv::imencode(".jpg", dataMat, buf);
+    return std::string(buf.begin(), buf.end());
```

Comment:
Here you create multiple copies - once from `dataMat` to `buf`, then from `buf` to return value